tests/test_cache.py

```diff
@@ -0,0 +1,130 @@
+    assert any(b['type'] == "wikipedia" for b in resp['blocks'][2].get('blocks'))
+
+    """
+    We make another request to the same POI which
+    should now be in the cache.
+    As a result no more Wikipedia call should be
+    made
+    """
+    response = client.get(
+       url=f'http://localhost/v1/pois/{louvre_museum}?lang=es',
+    )
+    resp = response.json()
+
+    assert len(mock_wikipedia.calls) == 2 # the same number of requests as before
+    assert any(b['type'] == "wikipedia" for b in resp['blocks'][2].get('blocks')) # we still have a wikipedia block
+
+
+def test_wikidata_cache(cache_test_normal, basket_ball, basket_ball_wiki_es, monkeypatch):
+    """
+    We test the cache for the Wikidata ES
+    """
+    client = TestClient(app)
+
+    with responses.RequestsMock(assert_all_requests_are_fired=False) as rsps:
+        """
+        We mock all wikipedia requests since
+        the information are expected to be in
+        the Wiki ES
+        """
+        rsps.add('GET',
+             re.compile('^https://.*\.wikipedia.org/'),
+             status=200)
+
+        response = client.get(
+            url=f'http://localhost/v1/pois/{basket_ball}?lang=fr',
+        )
+
+        assert response.status_code == 200
+        resp = response.json()
+        """
+        We should have a "wikipedia" block in the
+        answer
+        """
+        assert any(b['type'] == "wikipedia" for b in resp['blocks'][2].get('blocks'))
+
+        with monkeypatch.context() as m:
+            """
+            Now that the "basket ball" request should be
+            in the cache, we test that the same request
+            will not invoke the WikidataConnector
+
+            So we change the method used by the WikidataConnector
+            by a fake method to be sure the real method is not
+            called
+            """
+            from idunn.blocks.wikipedia import WikidataConnector
+
+            @wraps(WikidataConnector.get_wiki_info)
+            def fake_get_wiki_info():
+                """
+                Fake method for test
+
+                This method should never be called
+                """
+                raise Exception
+
+            m.setattr(WikidataConnector, "get_wiki_info", fake_get_wiki_info)
+
+            """
+            We make 10 requests to the basket_ball POI
+            and we should still have the wikipedia block
+            in the answer but without call to wikidata
+            neither wikipedia
+
+            Without the cache the request would fail
+            in the "get_wiki_info()" method
+            """
+            for i in range(10):
+                response = client.get(
+                    url=f'http://localhost/v1/pois/{basket_ball}?lang=fr',
+                )
+
+            assert len(rsps.calls) == 0 # Wikipedia API has never been called
+            assert any(b['type'] == "wikipedia" for b in resp['blocks'][2].get('blocks')) # we still have the wikipedia block
```

Comment:
This assert should be in the loop, and use the new `response` instead of the previous `resp` (that does not use the cache :wink:)